schema/descriptor.schema.json

```diff
@@ -28,18 +28,57 @@
             "description": "A string that describes the tool command line, where input and output values are identified by \"keys\". At runtime, command-line keys are substituted with flags and values.",
             "type": "string"
         },
-        "docker-image": {
-            "id": "http://github.com/boutiques/boutiques-schema/docker-image",
-            "minLength": 1,
-            "description": "Name of a Docker image where tool is installed and configured. Example: docker.io/neurodebian.",
-            "type": "string"
-        },
-        "docker-index": {
-            "id": "http://github.com/boutiques/boutiques-schema/docker-index",
-            "minLength": 1,
-            "description": "Docker index where Docker image is available.",
-            "default": "http://index.docker.io",
-            "type": "string"
+        "container": {
+            "id": "http://github.com/boutiques/boutiques-schema/container",
+            "type": "object",
+            "description": "Object representing a container for the tool.",
+            "allOf": [{
+                "properties": {
+                    "working-directory": {
+                        "id": "http://github.com/boutiques/boutiques-schema/container/working-directory",
+                        "minLength": 1,
+                        "description": "Location from which this task must be launched within the container.",
+                        "type": "string"
+                    },
+                    "container-hash": {
+                        "id": "http://github.com/boutiques/boutiques-schema/container/container-hash",
+                        "minLength": 1,
+                        "description": "Hash for the given container.",
+                        "type": "string"
+                    }
+                }
```

Comment:
Related to the earlier conversation: couldn't you move this `properties` key up to the `container` level? Then you could ditch the `allOf` -> `oneOf` nesting (you'd just need `oneOf`).